modules/nix/linux-builder.nix

```
@@ -7,9 +7,9 @@ let
 
   cfg = config.nix.linux-builder;
 
-  builderWithOverrides = cfg.package.override {
-    modules = [ cfg.config ];
-  };
+  builderWithOverrides = cfg.package.override (previousArguments: {
+    modules = previousArguments.modules ++ [ cfg.config ];
```

What happens if `previousArguments.modules` is null? Can that ever happen?

When I'm doing overrides like this, I generally use this pattern:
```suggestion
    modules = (previousArguments.modules or []) ++ [ cfg.config ];
```

---

If we know for a fact that `previousArguments.modules` will never be null, then feel free to ignore this suggestion (though a comment indicating that we know it can never be null might still be helpful)

---

Good question! I had to check to be sure. Yes, modules is a currently a required argument of the function used to create these packages, there is no default.  I'll add a comment.